app/src/ApiClient.ts

```diff
@@ -163,7 +163,12 @@ export interface HpkeConfig {
 const mime = "application/vnd.divviup+json;version=0.1";
 
 export class ApiClient {
-  private client?: Promise<AxiosInstance> | AxiosInstance;
+  #client: Promise<AxiosInstance>;
+  #currentUser?: Promise<User>;
+
+  constructor() {
+    this.#client = this.buildClient();
```

Comment:
previously we waited for the first api request to fetch the base url and build the client. now we initiate that promise immediately, increasing the likelihood that it will be resolved by the time we need it